### ISSUE TITLE
Use open-source Snorkel to create labelling functions to expand our training dataset.

### ISSUE THREAD

**Issue by bruffridge** [nasa-petal/PeTaL-labeller#65]
https://www.snorkel.org/get-started/

https://github.com/snorkel-team/snorkel

**Comment by bruffridge** [comment]
**Labelling functions**

Sense temperature cues: `'biology' AND ('temperature measurement' OR 'temperature monitoring' OR 'temperature sensing')`
Send light signals in the non-visible spectrum: `'biology' AND 'ultraviolet'`
Capture energy: `'biology' AND 'energy harvesting'`
Protect from wind: `'biology' AND ('wind stress' OR 'wind engineering')`

**Comment by bruffridge** [comment]
Another idea for a labelling function is to take our ground truth labelled papers, and use the MAG API to find papers related to each one. Each related paper gets assigned the same label as the ground truth paper it's related to.

Here's how to get related papers from MAG (kudos to @dsmith111 for figuring this out!):
> Passing the paper's Id into this link: https://academic.microsoft.com/api/entity/***?entityType=2
(Where *** is the Id) dumps all of the related paper Ids in a nice JSON format.

> Example link:
https://academic.microsoft.com/api/entity/3012421327?entityType=2
